src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/run.py

```diff
@@ -67,25 +76,52 @@ def acr_run(cmd,  # pylint: disable=too-many-locals
         source_location = check_remote_source_code(source_location)
         logger.warning("Sending context to registry: %s...", registry_name)
 
-    platform_os, platform_arch, platform_variant = get_validate_platform(cmd, os_type, platform)
-
-    FileTaskRunRequest, PlatformProperties = cmd.get_models('FileTaskRunRequest', 'PlatformProperties')
-    request = FileTaskRunRequest(
-        task_file_path=file,
-        values_file_path=values,
-        values=(set_value if set_value else []) + (set_secret if set_secret else []),
-        source_location=source_location,
-        timeout=timeout,
-        platform=PlatformProperties(
-            os=platform_os,
-            architecture=platform_arch,
-            variant=platform_variant
-        ),
-        credentials=get_custom_registry_credentials(
-            cmd=cmd,
-            auth_mode=auth_mode
+    if file == "-":
+        import sys
+        yaml_template = ""
+        for s in sys.stdin.readlines():
+            yaml_template += s
+        values_content = ""
+
+    if cmd_value:
+        yaml_template = "steps: \n  - cmd: {{ .Values.run_image }}\n"
```

Comment:
It just occurs to me the step default timeout is 10 minutes. If timeout is passed into the command, we need to add timeout https://github.com/Azure/acr-builder/blob/master/docs/task.md#steptimeout
https://github.com/Azure/acr-builder/blob/master/docs/task.md#totaltimeout

Reply:
Added timeout for each step in the yaml template.

Reply:
where did you add the timeout in the yaml template?

Reply:
Only did this for `az acr task create`. Now `az acr run` is having timeout in the yaml template too.